product/util.py

```diff
@@ -171,164 +211,197 @@ def diff_products(left_product, right_product, ignore_component_names=()):
     )
 
 
-@dataclasses.dataclass
-class ComponentDiff:
-    crefs_only_left: set = dataclasses.field(default_factory=set)
-    crefs_only_right: set = dataclasses.field(default_factory=set)
-    crefpairs_version_changed: set = dataclasses.field(default_factory=set)
-    names_only_left: set = dataclasses.field(default_factory=set)
-    names_only_right: set = dataclasses.field(default_factory=set)
-    names_version_changed: set = dataclasses.field(default_factory=set)
-
+def diff_components(
+    left_components: typing.Tuple[cm.Component],
+    right_components: typing.Tuple[cm.Component],
+    ignore_component_names=(),
+) -> ComponentDiff:
+    left_identities = {
```

Comment:
it might make the code easier to read if those were called component_identities (at the price of being veeeryyy long names..)

Reply:
i changed this for the sake of readability